### PR TITLE
added test to look for loops

### PR DESCRIPTION
**Pull request by ronpeer**

### CODE REVIEW

File: shlomobot_pytest/common_tests.py [hunk 1]
```diff
@@ -140,3 +137,25 @@ def declared_global_variable(file_list: list[str]) -> bool:
             return True
 
     return False
+
+
+def contains_loops(file_list: list[str], function_name_list: list[str]) -> bool:
```
Comment (KeshetS) [inline shlomobot_pytest/common_tests.py:142]: Better to split it into two functions. One that will check for for loops, and one for while loop.
In addition, the function currently returns True even when there are no loops in the code at all.

**Review by KeshetS — CHANGES_REQUESTED** [review]

File: shlomobot_pytest/common_tests.py (hunk 1 above)
Comment (maxandron) [reply shlomobot_pytest/common_tests.py:142]: Also consider what happens if any of the functions in the function_names is not contained in any module

File: shlomobot_pytest/common_tests.py [hunk 2]
```diff
@@ -2,22 +2,45 @@
...
+ABSOLUTE_PATH_REGEX_WINDOWS_PARAM = re.compile(
+    r'\((?:[^\s]*? ?, ?)*?[\'"][a-zA-Z]:.*\)\s*$'
+)
+WITH_OPEN_REGEX = re.compile(r"with\s+open\(['\"]")
+FOR_LOOP_REGEX = re.compile(r"for\s+\w+\s+in\s+\w+")
+LAMBDA_REGEX = re.compile(r"lambda (?:[^\s]*? ?, ?)*?\w+\s*:")
+ABSOLUTE_PATH_REGEX_UNIX = re.compile(r'\w+\s*=\s*[\'"]\/\w+')
+ABSOLUTE_PATH_REGEX_WINDOWS = re.compile(r'\w+\s*=\s*[\'"][a-zA-Z]:')
+ABSOLUTE_PATH_REGEX_UNIX_PARAM = re.compile(r'\((?:[^\s]*? ?, ?)*?[\'"]\/.*\)\s*$')
+GLOBAL_DECLERATION_REGEX = re.compile(r"\n\s*global\s+[^\W]+(?:\s*,\s*[^\W]+\s*)*\n")
```
Comment (KeshetS) [inline shlomobot_pytest/common_tests.py:33]: The regex doesn't work.

**Review by KeshetS — CHANGES_REQUESTED** [review]

File: shlomobot_pytest/common_tests.py (hunk 2 above)
Comment (kjdart) [reply shlomobot_pytest/common_tests.py]: Fixed in latest commit

File: shlomobot_pytest/common_tests.py [hunk 3]
```diff
@@ -160,7 +160,34 @@ def function_is_one_liner(py_filename: str, function_name: str) -> bool:
...
+        
+        a()
+    """
+    # Get instructions from the generated bytecode of the function
+    bytecode = dis.Bytecode(function)
+    instructions = [instruction for instruction in bytecode]
+
+    for line_number, instruction in enumerate(instructions):
+        # Search for a CALL_FUNCTION instruction
+        if instruction.opname != "CALL_FUNCTION":
```
Comment (KeshetS) [inline shlomobot_pytest/common_tests.py:176]: Instead of using != and continue, why not use == and put the rest of the relevant code lines here?

**Review by KeshetS — CHANGES_REQUESTED** [review]

**Review by KeshetS — APPROVED** [review]

**Review by KeshetS — APPROVED** [review]

File: shlomobot_pytest/common_tests.py (hunk 3 above)
Comment (kjdart) [reply shlomobot_pytest/common_tests.py]: Changed in latest commit

**Review by KeshetS — APPROVED** [review]

**Review by KeshetS — APPROVED** [review]

**Pull request merged**